csharp/src/Ice/Internal/EndpointFactoryManager.cs

```diff
@@ -24,6 +24,11 @@ public void add(EndpointFactory factory)
     {
         lock (_mutex)
         {
+            if (_destroyed)
```

Comment:
For C# and Java, it would be much simpler to remove destroy/_destroyed. destroy is about breaking cycles in C++, and we don't need that in C# and Java. On the contrary, it makes the code much more complicated and (like here) incorrect since it makes race conditions possible in the endpoint factory implementations.